File: 2918-minimum-equal-sum-of-two-arrays-after-replacing-zeros/2918-minimum-equal-sum-of-two-arrays-after-replacing-zeros.java
class Solution {
    public long minSum(int[] nums1, int[] nums2) {
        long sum1=0;
        long sum2=0;
        int zeros1=0;
        int zeros2=0;
        for (int i : nums1) {
            sum1+=i;
            if (i==0) {
                zeros1++;
            }
        }
        for (int i : nums2) {
            sum2+=i;
            if (i==0) {
                zeros2++;
            }
        }
        sum1+=zeros1;
        sum2+=zeros2;
        if (sum1==sum2) return sum1;
        if (sum1<sum2 && zeros1>0) return sum2;
        else if (sum1>sum2 && zeros2>0) return sum1;
        return -1;
            
        
        
    }
}

File: 2918-minimum-equal-sum-of-two-arrays-after-replacing-zeros/2918-minimum-equal-sum-of-two-arrays-after-replacing-zeros.cpp
class Solution {
public:
    long long minSum(vector<int>& nums1, vector<int>& nums2) {
        long long int sum1 = 0,sum2 = 0;
        int ct0_1 = 0,ct0_2 = 0;
        for(int i:nums1){
            sum1 += i;
            if(i == 0){
                ct0_1++;
            }
        }
        for(int j:nums2){
            sum2 += j;
            if(j == 0){
                ct0_2++;
            }
        }
        if(ct0_1 == 0 && (sum1 < sum2+ct0_2) || ct0_2 == 0 && (sum2 < sum1 + ct0_1)) return -1;

        return max((sum1+ct0_1),(sum2+ct0_2));
    }
};